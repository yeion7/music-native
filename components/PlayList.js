/* @flow weak */

import React from "react";
import { View, Dimensions } from "react-native";
import { Text, List, ListItem, Left, Right } from "native-base";
import styled from "styled-components/native";

const { width } = Dimensions.get("window");

import { formattedTime } from "../lib/time";
const PlayList = ({ playList, onPressSong }) => (
  <Wrapper>
    <List
      dataArray={playList}
      renderRow={song => (
        <ListItem key={song.id} onPress={() => onPressSong(song)}>
          <Left>
            <Text>{song.name}</Text>
          </Left>
          <Right>
            <Text>{formattedTime(song.duration_ms)}</Text>
          </Right>
        </ListItem>
      )}
    />
  </Wrapper>
);

const Wrapper = styled.View`
  width: ${width}
`;
export default PlayList;
