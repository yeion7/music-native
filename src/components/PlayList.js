/* @flow weak */

import React from "react";
import { View, Dimensions } from "react-native";
import { List } from "native-base";

import styled from "styled-components/native";
import ItemContainer from "../containers/ItemContainer";

const { width, height } = Dimensions.get("window");

const RenderItem = song => <ItemContainer {...song} key={song.id} />;

const PlayList = ({ playList }) => (
  <Wrapper>
    <List dataArray={playList} renderRow={RenderItem} />
  </Wrapper>
);

const Wrapper = styled.View`
  flex:1;
  width: ${width};
  height: ${height};
`;
export default PlayList;
