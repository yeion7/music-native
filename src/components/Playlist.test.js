import React from "react";
import "react-native";

import PlayList from "./PlayList";

import renderer from "react-test-renderer";

describe("renders test ", () => {
  it("render PlayList component", () => {
    const tree = renderer
      .create(
        <PlayList
          playList={[
            {
              id: "53Dj5PCDhb22qWqmre3YQs",
              name: "Aw Naw",
              duration_ms: "188706"
            }
          ]}
          onPressSong={jest.fn}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
