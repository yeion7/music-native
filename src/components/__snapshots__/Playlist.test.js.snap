// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders test  render PlayList component 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "height": 667,
        "width": 375,
      },
      undefined,
    ]
  }
>
  <ScrollView
    dataArray={
      Array [
        Object {
          "duration_ms": "188706",
          "id": "53Dj5PCDhb22qWqmre3YQs",
          "name": "Aw Naw",
        },
      ]
    }
    dataSource={
      ListViewDataSource {
        "items": 1,
      }
    }
    enableEmptySections={true}
    initialListSize={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={1000}
    onKeyboardDidHide={undefined}
    onKeyboardDidShow={undefined}
    onKeyboardWillHide={undefined}
    onKeyboardWillShow={undefined}
    onLayout={[Function]}
    onScroll={[Function]}
    pageSize={1}
    removeClippedSubviews={true}
    renderRow={[Function]}
    scrollEventThrottle={50}
    scrollRenderAheadDistance={1000}
    stickyHeaderIndices={Array []}
    stickySectionHeadersEnabled={true}
    style={Object {}}
    virtual={undefined}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": undefined,
            "borderBottomWidth": 0.5,
            "borderColor": "#c9c9c9",
            "flexDirection": "row",
            "marginLeft": 15,
            "paddingBottom": 13,
            "paddingRight": 15,
            "paddingTop": 13,
          }
        }
        virtual={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "alignSelf": "center",
              "flex": 1,
              "flexDirection": "row",
            }
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
                "marginLeft": 10,
              }
            }
            virtual={undefined}
          >
            Aw Naw
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "alignSelf": "center",
              "flex": 0.28,
              "paddingBottom": null,
              "paddingLeft": null,
              "paddingRight": null,
              "paddingTop": null,
            }
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": null,
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              }
            }
            virtual={undefined}
          >
            3:08
          </Text>
        </View>
      </View>
    </View>
  </ScrollView>
</View>
`;
